DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,400 @@
+                    return null;
+                }
+            }
+        }
+
+        private static List<ulong> BuildGlobalDeleteList(List<DiscordApplicationCommand> updateList = null)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any())
+            {
+                return null;
+            }
+            else if (updateList == null)
+            {
+                var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._globalDiscordCommands;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        if (!updateList.Any(ul => ul.Name == cmd.Name))
+                            InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        private static List<DiscordApplicationCommand> BuildGlobalCreateList(List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any())
+            {
+                return updateList;
+            }
+            else if (updateList == null)
+            {
+                return null;
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._globalDiscordCommands;
+                List<DiscordApplicationCommand> NewCommands = new();
+                foreach (var cmd in updateList)
+                {
+                    if (discord == null || !discord.Any(d => d.Name == cmd.Name))
+                    {
+                        NewCommands.Add(cmd);
+                    }
+                }
+
+                return NewCommands;
+            }
+        }
+
+        private static Dictionary<ulong, DiscordApplicationCommand> BuildGlobalOverwriteList(List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._globalDiscordCommands == null || !ApplicationCommandsExtension._globalDiscordCommands.Any())
+            {
+                return null;
+            }
+            else if (updateList == null)
```

Comment:
This condition could be combined with the first one. Further reducing the branching being done here